cpp/src/arrow/util/byte_stream_split_internal.h

```diff
@@ -117,20 +122,22 @@ void ByteStreamSplitEncodeSse2(const uint8_t* raw_values, const int64_t num_valu
   // The current shuffling algorithm diverges for float and double types but the compiler
   // should be able to remove the branch since only one path is taken for each template
   // instantiation.
-  // Example run for floats:
+  // Example run for 32-bit variables:
   // Step 0, copy:
   //   0: ABCD ABCD ABCD ABCD 1: ABCD ABCD ABCD ABCD ...
-  // Step 1: _mm_unpacklo_epi8 and mm_unpackhi_epi8:
+  // Step 1: simd_batch<int8_t, 8>::xip_lo and simd_batch<int8_t, 8>::xip_hi:
   //   0: AABB CCDD AABB CCDD 1: AABB CCDD AABB CCDD ...
   //   0: AAAA BBBB CCCC DDDD 1: AAAA BBBB CCCC DDDD ...
```

Comment:
Can you add the missing comment for step 2?

Reply:
Aha, there is no step 2, let me edit the number here.

Reply:
There are two lines above, so clearly there must be a transformation between them?

Reply:
So there really are 2 steps? Calling it "stage" doesn't change anything.

Reply:
"stage" is variable name here. It use `xsimd::zip_lo, xsimd::zip_hi` twice to build the result here, I don't know what naming could be better here...🤔

Reply:
Please re-read.

Reply:
I've try to update it, would you mind take a look again?

Reply:
+1, thank you!